tf.image.per_image_standardization respects input dtype

chrisyeh96 opened the pull request:
Make it clear that it can operate on batches of image

chrisyeh96 committed:
Clarify tf.image.per_image_standardization documetation

Make it clear that it can operate on batches of image

gbaned commented:
@chrisyeh96 can you please check ubuntu sanity errors ?

chrisyeh96 committed:
per_image_standardization respects input dtype

chrisyeh96 commented:
Whoops, turns out I went over the 80-chars-per-line limit. I also took this opportunity to make the `tf.image.per_image_standardization` respect the input image's dtype. Please review. Thank you!

alextp reviewed (changes requested):
Can you add a unit test which fails before this change but passes after?

chrisyeh96 commented:
where should I write the unit tests?

alextp commented:
tensorflow/python/ops/image_ops_test.py

On Wed, May 1, 2019 at 4:30 PM Christopher Yeh <notifications@github.com>
wrote:

> where should I write the unit tests?
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/tensorflow/tensorflow/pull/28210#issuecomment-488487838>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAABHROKNFLRUQFRCYB7QRTPTIRXTANCNFSM4HI34UMQ>
> .
>


-- 
 - Alex

gbaned commented:
@chrisyeh96 Did you get a chance to look on reviewer comments. Please let us know on the update. Thanks!

gbaned commented:
@chrisyeh96 gentle ping to address reviewer comments. Thanks!

chrisyeh96 committed:
tf.image.per_image_standardization: unit test for preserving dtype

chrisyeh96 commented:
Just added unit tests! @alextp

alextp commented on tensorflow/python/ops/image_ops_test.py:
```diff
@@ -1494,6 +1493,18 @@ def testBatchWhitening(self):
+  def testPreservesDtype(self):
+    imgs_npu8 = np.random.uniform(0., 255., [2, 5, 5, 3]).astype(np.uint8)
+    imgs_npf16 = np.random.uniform(0., 255., [2, 5, 5, 3]).astype(np.float16)
+    with self.cached_session(use_gpu=True):
```
You don't need a session here

chrisyeh96 replied:
good catch. fixed

alextp reviewed (changes requested):
Thanks!

chrisyeh96 committed:
Remove unnecessary session creation in per_image_standardization unit test

chrisyeh96 committed:
Remove unused import in image_ops_impl.py

chrisyeh96 commented:
@alextp It ran into linter errors, so I fixed those and rebased master as well (hence the force-push). If you can take another look, that would be great. Thanks!

chrisyeh96 committed:
Fix typo

chrisyeh96 committed:
image_ops_impl.py: fix dtype mismatch, correct typos

chrisyeh96 commented:
@alextp I don't think the Windows build failure is due to any of the changes I introduced. What does the pull-request process look like from here on out?

alextp commented:
@chrisyeh96 we'll try to submit until CI passes

chrisyeh96 commented:
> 
> 
> @chrisyeh96 we'll try to submit until CI passes

@alextp Do the internal CI builds automatically re-run every time new commits are merged into master? Or is there someone (perhaps yourself) manually requesting that the CI runs?

chrisyeh96 commented:
@alextp are there any updates on this? Thanks!

alextp commented:
Still retrying... :-/

chrisyeh96 commented:
Thanks @alextp! Really appreciate it!
